In [1]:
import os
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

# HW 5

In [9]:
train = pd.read_csv("MusicRatingsTrain.csv")
test = pd.read_csv("MusicRatingsTest.csv")
vala = pd.read_csv("MusicRatingsValidationA.csv")
valb = pd.read_csv("MusicRatingsValidationB.csv")
train.head()

userID  songID    rating                     songName  year  \
0    2071     581  2.484754                 Show You How  2004   
1    1208      55  1.000000                    Luvstruck  1999   
2    1484     510  1.000000                   In One Ear  2008   
3     453     299  1.000000  Gardenhead / Leave Me Alone  1992   
4    2238     113  1.000000                      Hustler  2006   

                artist       genre  
0          The Killers        Rock  
1   Southside Spinners  Electronic  
2    Cage The Elephant        Rock  
3   Neutral Milk Hotel        Rock  
4  Simian Mobile Disco  Electronic

# Question 1

# a

In [13]:
songs = len(pd.unique(train['songID']))
songs

807

In [14]:
users = len(pd.unique(train['userID']))
users

2421

In [16]:
stats = train['rating'].describe()
stats

count    245997.000000
mean          1.321962
std           0.459894
min           1.000000
25%           1.000000
50%           1.000000
75%           1.494918
max           4.768656
Name: rating, dtype: float64

In [18]:
median = train['rating'].median()
median

1.0

# b) i.

There are a total of 807 + 2421 = 3228 parameters included in the model. We have to train the model with 2421 observations because there are no null ratings in the training dataset, so the number of observations is the same as the number of ratings.

# b) ii.

In [23]:
from fancyimpute import BiScaler

ModuleNotFoundError: No module named 'fancyimpute'